src/app/component/shared/secondary-menu/secondary-menu.component.ts

```diff
@@ -50,8 +51,26 @@ export class SecondaryMenuComponent implements OnInit, OnDestroy {
         while (menuRoute.parent) {
           menuRoute = menuRoute.parent;
           parentMenuRoutes.push(menuRoute);
+          indentation++;
         }
 
+        // Update indentation values
+        if (current.order) {
+          current.order.indentation = indentation;
+        } else {
+          current.order = { indentation };
+        }
+
+        parentMenuRoutes.forEach(parent => {
```

Comment:
it isn't readily obvious why you have to set indentation on parent routes?

This loop has the potential to run many times...